packages/plugin-ext/src/plugin/document-data.ts

```diff
@@ -189,7 +189,7 @@ export class DocumentDataExt {
         }
 
         if (range.isSingleLine) {
-            return this.lines[range.start.line].substring(range.start.character, range.end.character);
+            return this.lines[range.start.line - 1].substring(range.start.character, range.end.character);
```

Comment:
do we have some tests for the range ?

Reply:
AFAIK we do not have any test for API so far